src/box/lua/config/instance_config.lua

```diff
@@ -653,25 +656,49 @@ return schema.new('instance_config', schema.record({
         end,
     }),
     iproto = schema.record({
-        -- XXX: listen/advertise are specific: accept a string of
-        -- a particular format, a number (port), a table of a
-        -- particular format.
-        --
-        -- Only a string is accepted for now.
-        listen = schema.scalar({
-            type = 'string',
+        listen = schema.array({
+            items = schema.record({
+                uri = schema.scalar({
+                    type = 'string',
+                    validate = function(data, w)
+                        -- Substitute variables with placeholders to don't
+                        -- confuse the URI parser with the curly brackets.
+                        data = data:gsub('{{ *.- *}}', 'placeholder')
+                        local uris, err = urilib.parse_many(data)
+                        if uris == nil then
+                            w.error('Unable to parse an URI/a list of URIs: %s',
+                                    err)
+                        end
+                    end,
+                }),
+                params = schema.record({
+                    transport = schema.enum({
+                        'plain',
+                        'ssl',
+                    }, {
+                        default = 'plain',
+                    }),
+                    ssl_ca_file = enterprise_edition(schema.scalar({
+                        type = 'string',
+                    })),
+                    ssl_cert_file = enterprise_edition(schema.scalar({
+                        type = 'string',
+                    })),
+                    ssl_ciphers = enterprise_edition(schema.scalar({
+                        type = 'string',
+                    })),
+                    ssl_key_file = enterprise_edition(schema.scalar({
+                        type = 'string',
+                    })),
```

Comment:
What is important about all the file paths in the configuration: it should point to the same file before and after the first `box.cfg()` call even if the path is relative and `process.work_dir` is set.

I don't know, whether `box.cfg()` handles it on its own and, if so, how exactly.

Please, check this point. If a special handing is required, consider generalizing code from `safe_mkdir()` (`applier/mkdir.lua`).

Reply:
We checked that there were no problems and all relative `ssl_*_file` paths were found in `process.work_dir`. There is, however, no test for now.

Reply:
I'm OK to leave it as `box.cfg()` deal.